custom_components/tuya_local/devices/vivosun_de0003_dehumidifier.yaml

```diff
@@ -0,0 +1,185 @@
+name: Dehumidifier
+products:
+  - id: qrjffdu2bjrd7v42
+    name: DE0003 10L Smart Dehumidifier
+    manufacturer: Vivosun
+primary_entity:
+  entity: humidifier
+  class: dehumidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: "mdi:air-humidifier-off"
+          icon_priority: 3
+        - dps_val: true
+          icon: "mdi:air-humidifier"
+          icon_priority: 5
+    - id: 2
+      type: integer
+      name: humidity
+      range:
+        min: 30
+        max: 80
+      mapping:
+        - step: 5
+    - id: 5
+      name: mode
+      type: string
+      mapping:
+        - dps_val: Continuities
+          value: boost
+          icon: "mdi:tshirt-crew-outline"
+          icon_priority: 4
+        - dps_val: Auto
+          value: auto
+          icon: "mdi:water-outline"
+          icon_priority: 4
+        - dps_val: Sleep
+          value: sleep
+          icon: "mdi:weather-night"
+          icon_priority: 4
+    - id: 6
+      type: integer
+      name: current_humidity
+    # This doesn't actually have an ionizer, but having the dp
+    # will aid in detection
+    - id: 10
+      type: boolean
+      name: ionizer
+    - id: 19
+      name: error
+      type: bitfield
+      # E2 might be a compressor fault.
+      mapping:
+        - dps_val: 0
+          value: OK
+        - dps_val: 1
+          value: "Water Tank Full or Removed"
+          icon: "mdi:cup-water"
+          icon_priority: 1
+        - dps_val: 2
+          value: "E2 Compressor Fault"
+          icon: "mdi:engine-off-outline"
+          icon_priority: 1
+secondary_entities:
+  - entity: fan
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+      - id: 4
+        type: string
+        name: speed
+        mapping:
+          - dps_val: "low"
+            value: 50
+          - dps_val: "high"
+            value: 100
+  - entity: sensor
```

Comment:
Don't add deprecated entities to new configs, they are only there in the existing ones to avoid breaking existing setups from before the current humidity was added to the humidifier entity.

Reply:
Done, I should have looked at #1408 more closely.